src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -948,7 +948,17 @@ public void spawnCitizen(final CitizenData data)
             CitizenData citizenData = data;
             if (citizenData == null)
             {
-                topCitizenId++;
+                /**This ensures that citizen IDs are getting reused.
+                 * That's needed to prevent bugs when calling IDs that are not used.
+                 */
+                for (int i = 1; i <= citizenData.getColony().getMaxCitizens(); i++)
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'citizenData' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2259)